server/api/utils/pagination.py

```diff
@@ -79,7 +80,7 @@ async def paginate_permission_filtered_request(
                 page_size,
                 **method_kwargs,
             )
-            new_result = await filter_(new_result)
+            new_result = await self._call_or_await_method(False, filter_, new_result)
```

Comment:
why False?

Reply:
Because I didn't believe there'd be a sync filter I need to run in a thread pool but I guess better be safe than sorry